tests/profiling/collector/test_memalloc.py

```diff
@@ -1,23 +1,34 @@
+from __future__ import annotations
+
 import gc
 import inspect
 import os
 from pathlib import Path
 import sys
 import threading
+from tracemalloc import Statistic
+from typing import TYPE_CHECKING
+from typing import Callable
+from typing import Sequence
 
 import pytest
+from typing_extensions import Union
 
 from ddtrace.internal.datadog.profiling import ddup
 from ddtrace.profiling.collector import memalloc
 from tests.profiling.collector import pprof_utils
 
 
+if TYPE_CHECKING:
+    from tests.profiling.collector import pprof_pb2  # pyright: ignore[reportMissingModuleSource]
+
+
 PY_314_OR_ABOVE = sys.version_info[:2] >= (3, 14)
 PY_313_OR_ABOVE = sys.version_info[:2] >= (3, 13)
 PY_311_OR_ABOVE = sys.version_info[:2] >= (3, 11)
 
 
-def _allocate_1k():
+def _allocate_1k() -> list[object]:
```

Comment:
list --> List

Reply:
> List (capitalized): This is a generic alias provided by the typing module (from typing import List). It was created specifically to allow type hinting with subscripting before Python 3.9, as the built-in list class did not originally support this syntax. It is now considered deprecated (though still functional) if you are targeting Python 3.9 or newer


Well, TIL. I believe we support v3.8 still, so maybe should do this - unless it's not true for `memalloc` ?